sdk/storage/azure-storage-common/src/test/java/com/azure/storage/common/implementation/StorageCrc64CalculatorTests.java

```diff
@@ -0,0 +1,174 @@
+
+public class StorageCrc64CalculatorTests {
+    private static final int KB = 1024;
+    private static final String HEX_DATA = "C8E11B40D793D1526018"; // '\\xC8\\xE1\\x1B\\x40\\xD7\\x93\\xD1\\x52\\x60\\x18'
+    private static final byte[] HEX_BYTES = hexStringToByteArray();
+
+    @ParameterizedTest
+    @CsvSource({
+        "'', 0, 0",
+        "'Hello World!', 0, 208604604655264165",
+        "'123456789!@#$%^&*()', 0, 2153758901452455624",
+        "'This is a test where the data is longer than 64 characters so that we can test that code path.', 0, 2736107658526394369", })
+    void testCompute(String data, long initial, long expected) {
+        byte[] bytes = data.getBytes();
+        long actual = StorageCrc64Calculator.compute(bytes, initial);
+        assertEquals(expected, actual);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+        "0, 3386042136331673945",
+        "208604604655264165, 4570059697646401418",
+        "2153758901452455624, 13366433516720813220",
+        "12345, 5139183895903464380" })
+    void testComputeWithBinaryData(long initial, String expectedStr) {
+        //byte[] hexBytes = hexStringToByteArray(hexData);
+        long expected = unsignedLongFromString(expectedStr);
+        long actual = StorageCrc64Calculator.compute(HEX_BYTES, initial);
+        assertEquals(expected, actual);
+    }
+
+    private static byte[] hexStringToByteArray() {
+        //int length = StorageCrc64CalculatorTests.hexData.length();
+        int length = HEX_DATA.length();
+        byte[] data = new byte[length / 2];
+        for (int i = 0; i < length; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(StorageCrc64CalculatorTests.HEX_DATA.charAt(i), 16) << 4)
+                + Character.digit(StorageCrc64CalculatorTests.HEX_DATA.charAt(i + 1), 16));
+        }
+        return data;
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = { 2, 3, 10 })
+    void testCompose(int numSegments) {
+        int blockSize = KB;
+        byte[] data = new byte[numSegments * blockSize];
+        new Random().nextBytes(data);
+
+        long wholeCrc = StorageCrc64Calculator.compute(data, 0);
+        Queue<Long> blockCrcs = new LinkedList<>();
+        IntStream.range(0, numSegments)
+            .forEach(i -> blockCrcs
+                .add(StorageCrc64Calculator.compute(Arrays.copyOfRange(data, i * blockSize, (i + 1) * blockSize), 0)));
+
+        long composedCrc = blockCrcs.poll();
+        int i = 1;
+        while (!blockCrcs.isEmpty()) {
+            long nextBlockCrc = blockCrcs.poll();
+            composedCrc
+                = StorageCrc64Calculator.concat(0, 0, composedCrc, ((long) blockSize * i), 0, nextBlockCrc, blockSize);
+            i++;
+        }
+
+        assertEquals(wholeCrc, composedCrc);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = { 2, 3, 10 })
+    void testComposeDifferingBlockSizes(int numSegments) {
+        int minBlockSize = KB, maxBlockSize = 4 * KB;
```

Comment:
It would be interesting if we could add test cases where we expand the block size to megabytes as I believe that will be the rough order of magnitude of individual checksums is used in the structured message.